packages/transaction-controller/src/helpers/PendingTransactionTracker.ts

```diff
@@ -286,7 +291,7 @@ export class PendingTransactionTracker {
 
     if (!rawTx?.length) {
```

Comment:
On further analysis, I think this is a deeper issue caused by the V2 base controller upgrade.

This approve should only be triggered if it has no `rawTx` implying it never went through the approve process which should rarely happen, if ever.

I think the issue is we add the `rawTx` property in `signTransaction` and then update the metadata, but then the code in `approveTransaction` doesn't update it's reference to the transaction metadata, meaning when it next updates to add the `hash` and change the status to `submitted`, it replaces the `rawTx` with `undefined`, and then hits this condition in the tracker.

I think the quickest and maybe cleanest fix would be to change the update on line 2748 to use `#updateTransactionInternal` so it can update only specific fields and therefore leave the `rawTx` in the metadata.